src/ClassicUO.Assets/TileDataLoader.cs

```diff
@@ -89,86 +89,93 @@ public unsafe Task Load()
                     _landData = new LandTiles[ArtLoader.MAX_LAND_DATA_INDEX_COUNT];
                     _staticData = new StaticTiles[staticscount * 32];
 
-                    byte* bufferString = stackalloc byte[20];
-
-                    for (int i = 0; i < 512; i++)
+                    if (isold)
                     {
-                        tileData.Skip(4);
-
-                        for (int j = 0; j < 32; j++)
+                        for (int i = 0, idx = 0; i < LAND_SIZE; i++)
                         {
-                            if (tileData.Position + (isold ? 4 : 8) + 2 + 20 > tileData.Length)
-                            {
-                                goto END;
-                            }
+                            tileData.Skip(4);
 
-                            int idx = i * 32 + j;
-                            ulong flags = isold ? tileData.ReadUInt() : tileData.ReadULong();
-                            ushort textId = tileData.ReadUShort();
+                            var landTilesPtr = (LandTilesOld *)tileData.PositionAddress;
+                            tileData.Skip(Marshal.SizeOf<LandTilesOld>() * 32);
 
-                            for (int k = 0; k < 20; ++k)
+                            for (int j = 0; j < 32; j++, idx++, landTilesPtr++)
                             {
-                                bufferString[k] = tileData.ReadByte();
+                                string name = string.Intern(Encoding.UTF8.GetString(landTilesPtr->Name, 20).TrimEnd('\0'));
+
+                                LandData[idx] = new LandTiles(landTilesPtr->Flags, landTilesPtr->TexID, name);
                             }
+                        }
 
-                            string name = string.Intern(Encoding.UTF8.GetString(bufferString, 20).TrimEnd('\0'));
+                        for (int i = 0, idx = 0; i < staticscount; i++)
+                        {
+                            tileData.Skip(4);
 
-                            LandData[idx] = new LandTiles(flags, textId, name);
+                            var staticTilesPtr = (StaticTilesOld *)tileData.PositionAddress;
+                            tileData.Skip(Marshal.SizeOf<StaticTilesOld>() * 32);
```

Comment:
Should we pull sizeof out as a variable? Not sure if this is a compile time constant or not.